src/python/pants/backend/awslambda/python/awslambda_python_rules.py

```diff
@@ -48,10 +53,23 @@ class LambdexSetup:
 
 @rule(desc="Create Python AWS Lambda", level=LogLevel.DEBUG)
 async def create_python_awslambda(
-    field_set: PythonAwsLambdaFieldSet, lambdex_setup: LambdexSetup
+    field_set: PythonAwsLambdaFieldSet, lambdex_setup: LambdexSetup, global_options: GlobalOptions
 ) -> CreatedAWSLambda:
     # Lambdas typically use the .zip suffix, so we use that instead of .pex.
-    pex_filename = f"{field_set.address.target_name}.zip"
+    disambiguated_pex_filename = os.path.join(
+        field_set.address.spec_path.replace(os.sep, "."), f"{field_set.address.target_name}.zip"
+    )
+    if global_options.options.pants_distdir_legacy_paths:
+        pex_filename = f"{field_set.address.target_name}.zip"
+        logger.warning(
+            f"Writing to the legacy subpath: {pex_filename}, which may not be unique. An "
+            f"upcoming version of Pants will switch to writing to the fully-qualified subpath: "
+            f"{disambiguated_pex_filename}. You can effect that switch now (and silence this "
+            f"warning) by setting `pants_distdir_legacy_paths = false` in the [GLOBAL] section of "
+            f"pants.toml."
+        )
```

Comment:
Yeah, I wasn't sure about using that, because I don't think there will ever be an error here. As long as the option exists, we will respect it, and once the option is removed, this code will be removed.